src/coreComponents/functions/TableFunction.cpp

```diff
@@ -180,6 +180,49 @@ void TableFunction::checkCoord( real64 const coord, localIndex const dim ) const
                  SimulationError );
 }
 
+void TableFunction::print( std::string const & filename ) const
+{
+  integer const numDimensions = LvArray::integerConversion< integer >( m_coordinates.size() );
+  GEOS_ERROR_IF( numDimensions > 2, "Printing tables with more than 2 dimensions is not supported" );
+
+  std::ofstream os( filename + ".csv" );
+
+  if( numDimensions == 1 )
+  {
+    arraySlice1d< real64 const > const coords = m_coordinates[0];
+    integer const N = coords.size();
+    os << units::getDescription( getDimUnit( 0 )) << "," << units::getDescription( m_valueUnit ) << "\n";
+    for( integer i = 0; i < N; i++ )
+    {
+      os << coords[i] << "," << m_values[i] << "\n";
+    }
+  }
+  else if( numDimensions == 2 )
+  {
+    arraySlice1d< real64 const > const coords0 = m_coordinates[0];
+    arraySlice1d< real64 const > const coords1 = m_coordinates[1];
+    integer const N0 = coords0.size();
+    integer const N1 = coords1.size();
+    os<<units::getDescription( getDimUnit( 0 ));
+    for( integer j = 0; j < N1; j++ )
+    {
+      os << "," << units::getDescription( getDimUnit( 1 )) << "=" << coords1[j];
+    }
+    os << "\n";
+    for( integer i = 0; i < N0; i++ )
+    {
+      os << coords0[i];
+      for( integer j = 0; j < N1; j++ )
+      {
+        os << "," << m_values[ j*N0 + i ];
+      }
+      os << "\n";
+    }
```

Comment:
I see that you went for the 2D layout of the table, with coordinates spread across both rows and columns. It's a valid choice, but does not generalize above 2D. If we ever extend this to arbitrary dimensions (printing at least 3D tables might be useful), I would prefer if all tables followed the same format:
```
NameX [UnitX],NameY [UnitY],NameZ [UnitZ],NameV [UnitV]
x[0],y[0],z[0],v[0][0][0]
x[1],y[0],z[0],v[1][0][0]
x[2],y[0],z[0],v[2][0][0]
...
x[0],y[1],z[0],v[0][1][0]
x[1],y[1],z[0],v[1][1][0]
...
x[NX-1],y[NY-1],z[NZ-1],v[NX-1][NY-1][NZ-1]
```
This is easy to consume in any tool (Excel, Matlab) and make plots since you can assign different columns to different axis. The order of coordinates/values doesn't really matter, but "Fortran" order feels most natural here.

Reply:
ok, i kept that layout for 2d tables but for all other dimensions it's your way
hope it's ok